rdoasis/algorithms/migrations/0008_alter_algorithm_id_alter_algorithmtask_id_and_more.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 4.0.3 on 2022-04-07 17:49
```

Comment:
These were generated with Django4 😉

Reply:
Yup. More specifically, those migrations were introduced by a recent version of `django-composed-configuration`. It seems what was happening is that since `django-composed-configuration` isn't pinned, `tox` was installing the latest version. And just to spice things up, the latest composed-configuration requires Django 4, AND the RGD versions pinned here were from before Django 4 support was added :)

So, I've pinned `django-composed-configuration` to the latest version, pinned Django so at least 4.0 is required, and also pinned all RGD dependencies to the latest version.